.github/workflows/job_test-suite.yml

```
@@ -0,0 +1,42 @@
+name: Run Test Suite
+
+on:
+  workflow_call:
+    inputs:
+      WORKING_DIRECTORY:
+        description: Path to working directory in repo
+        required: true
+        type: string
+
+jobs:
+  run-test-suite:
+    name: Run test suite and upload coverage artifact
+    runs-on: ubuntu-24.04
+    defaults:
+      run:
+        shell: bash
+        working-directory: ${{ inputs.WORKING_DIRECTORY }}
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          submodules: true
+
+      - name: Setup nodeJS v20
```

similar here on node version
```suggestion
      - name: Setup nodeJS
```